org.eclipse.lyo.oslc4j.provider.jena/src/main/java/org/eclipse/lyo/oslc4j/provider/jena/JenaModelHelper.java

```diff
@@ -2210,4 +2240,17 @@ private static Transformer createTransformer()
 		}
 	}
 
+    private static Class getResourceClass(Resource resource, Map<String, Class> rdfTypeClasses) {
+        String typeURI;
+        Statement statement;
+        Class foundType = null;
+        StmtIterator types = resource.listProperties(RDF.type);
+        while(types.hasNext() && foundType == null) {
+            statement = types.nextStatement();
+            typeURI = statement.getObject().asResource().getURI();
+            foundType = rdfTypeClasses.get(typeURI);
```

Comment:
is it really the strategy we want to have to resolve the _first_ RDF class to an existing Java class?

Reply:
A little more complex approach would be to allow the user to define preferred types, some how the same way he/she will define the packages to scan.